include/seqan3/utility/type_inspection.hpp

```diff
@@ -0,0 +1,68 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides traits to inspect some information of a type, for example its name.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#if defined(__GNUC__) || defined(__clang__)
+#include <cxxabi.h>
+#endif // defined(__GNUC__) || defined(__clang__)
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <typeinfo>
+
+#include <seqan3/core/platform.hpp>
```

Comment:
Hm, since utility should be independent from seqan, do we still need to include the platform header?

Reply:
Yes, they are not excluded from the header test.